### PR TITLE
feat: [PIE-6642]: add logic to work with gitClient for SSH type git connector

### PR DESCRIPTION
**Pull request by shalini-agr**
## Describe your changes

**What is the issue?**
To make SSH connector also work with gitClient.

**What is the change?**
Parsing url using slug and endpoint. in case of update API, first checking if the file exists or not and that commitId should be equal to latest on that branch.

Impact?
None

Testing?
a) If Manual -> create/update remote pipeline with gitClient enabled in settings 
b) If indexes -> No
c) If particular/complete sanity ran -> No

ReleaseNotes summary needed? No

Is it new task type? No
Did you make changes to the code in a backward-compatible manner? No
Did you add a new class which is added in kryo serialization? - No
Will the new version of the delegate receive old version task and process? No
If new version of the pipeline-service receives response for the old task will it be able to handle it successfully? No
Did you make a change in delegate task which will be created in pipeline-services via grpc call to manager? No
If the older version of pipeline-service creates the task parameter and calls newer cg manager, will it be okay? No
if the newer version of pipeline-service creates the task parameter and calls older cg manager, will it be okay? No
Did you remove an enum → No
Did you add an enum(That too at end of list) → No
FeatureFlag Added/Removed -> No
Verify Changes with affects any -> No

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/40804)

### CODE REVIEW

**Comment by shalini-agr** [comment]
trigger codebasehash

**Comment by mohitg0795** [comment]
*[`product/ci/scm/file/file.go` line 682 at r4](https://reviewable.io/reviews/harness/harness-core/40804#-NI99swU0X1jvStkKMoz:-NI99swVDmv48XBr8b8m:b-rixx3j) ([raw file](https://github.com/harness/harness-core/blob/07ff1542dde93678b1b3f2de8279a2e1cff3fd9e/product/ci/scm/file/file.go#L682)):*
> ```Go
> 
> func parseUrl(endpoint string, slug string) string {
> 	fmt.Println(endpoint + "scm/" + slug + ".git")
> ```

remove this print statement

**Review by mohitg0795 — COMMENTED** [review]
> Reviewed 1 of 2 files at r1, 1 of 2 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40804)* status: 3 file(s) to review. 0 of 2 approvals obtained.

**Comment by shalini-agr** [comment]
*[`product/ci/scm/file/file.go` line 682 at r4](https://reviewable.io/reviews/harness/harness-core/40804#-NI99swU0X1jvStkKMoz:-NIAm_f1-CO8scjWlaXQ:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/07ff1542dde93678b1b3f2de8279a2e1cff3fd9e/product/ci/scm/file/file.go#L682)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

remove this print statement
</blockquote></details>

Done.

File: product/ci/scm/proto/scm.proto [hunk 1]
```diff
@@ -265,7 +265,6 @@ message Provider {
   // If skipVerify is false, we can pass in a path of additional certs to allow
   // This might be useful in self hosted repositories which use self signed certs.
   string additional_certs_path = 10;
-  string url = 12;
```
Comment (shubham149) [inline product/ci/scm/proto/scm.proto:268]: This is a backward incompatible change. We cannot remove an attribute from api.

Comment (shalini-agr) [reply product/ci/scm/proto/scm.proto:268]: This was added by us only in the last PR, but now we are not using url anymore and generating it using endpoint and slug itself.

**Review by mohitg0795 — APPROVED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40804)* status: 3 file(s) to review. 1 of 2 approvals obtained.

File: product/ci/scm/proto/scm.proto (hunk 1 above)
Comment (shubham149) [reply product/ci/scm/proto/scm.proto:268]: Did the earlier change that included url reached prod?

Comment (shalini-agr) [reply product/ci/scm/proto/scm.proto:268]: I don’t think so, we only merged it yesterday.

**Review by shubham149 — APPROVED** [review]
> Reviewed 1 of 2 files at r1, 1 of 1 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40804)* status: :shipit: complete! ✓

**Pull request merged**